pkg/plugin/plugin.go

```diff
@@ -78,6 +82,55 @@ type Info struct {
 	ConfigAttributes json.RawMessage `db:"config_attrs" json:"config_attrs"` // ConfigOption(s) as json-encoded list
 }
 
+// NewInfo constructs a new Info to be used in Plugin.GetInfo.
+//
+// The configOptions will be enriched with defaults from the typePtr, being a pointer to the Plugin implementation.
+func NewInfo(typePtr interface{}, name, version, author string, configOptions []*ConfigOption) (*Info, error) {
```

Comment:
Shouldn't the type of the `typePtr` argument always be `Plugin` or have I missed something?

Reply:
Fair enough. I haven't set the type to `Plugin` in the first place as none of the `Plugin` interface methods were used, but it makes the API clearer.